File: SLMS_UI/src/app/core/models/CreateBranchRequest.ts
import { InstitutionStatus } from "@core/enums/OnbardingSteps";

export interface CreateBranchRequest {
  name: string;
  institutionId: string; // Guid
  email?: string;
  phone?: string;
  description?: string;
  address?: string;
  city?: string;
  closesAt?: string; // TimeOnly (e.g., "18:00")
  openAt?: string;   // TimeOnly (e.g., "09:00")
  capacity?: number;
  isActive: boolean;
  isPrimary: boolean;
  isOnboarding: boolean;
  status : InstitutionStatus
}

File: SLMS_API/Application/Contracts/Organizations/Requests/CreateBranchRequest.cs
using SLMS_API.Common.Enums;

namespace SLMS_API.Application.Contracts.Organizations.Requests;

public class CreateBranchRequest
{
    public string Name { get; set; } = string.Empty;
    public Guid InstitutionId { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public string? Description { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public TimeOnly? ClosesAt { get; set; }
    public TimeOnly? OpenAt { get; set; }
    public int? Capacity { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsPrimary { get; set; } = true;
    public bool IsOnboarding { get; set; } = false;
    public InstitutionStatus Status { get; set; } = InstitutionStatus.Active;
}
